src/AzureClient/AzureClient.cs

```diff
@@ -572,10 +585,31 @@ public async Task<ExecutionResult> GetJobResultAsync(IChannel? channel, string j
                 //                 cancellation token support.
                 var request = WebRequest.Create(job.OutputDataUri);
                 using var responseStream = (await request.GetResponseAsync()).GetResponseStream();
-                return responseStream.ToHistogram(
-                    Logger, 
-                    isSimulatorOutput: this.ActiveTarget?.TargetId?.StartsWith(MicrosoftSimulator) ?? false)
-                .ToExecutionResult();
+                if (this.ActiveTarget?.TargetId?.StartsWith(MicrosoftSimulator) ?? false)
+                {
+                    var outputLines = ReadAllLines(responseStream).ToList();
+                    outputLines = outputLines.Select(line => line.Trim()).ToList();
```

Comment:
Given that you return an `IEnumerable`, I believe you could just concatenate linq calls without having to use `ToList()`:
```csharp
var outputLines = ReadAllLines(responseStream)
    . Select(line => line.Trim())
    .ToList();
```

Reply:
I merged the ReadAllLines content to inside of the function and just did the line trimming there.